### PR TITLE
chore(release): Publish packages

### PR DESCRIPTION
**Pull request by spydon**
- audioplayers@3.0.0
 - audioplayers_android@2.0.0
 - audioplayers_darwin@3.0.0
 - audioplayers_linux@1.0.4
 - audioplayers_platform_interface@4.0.0
 - audioplayers_web@2.2.0
 - audioplayers_windows@1.1.3

### CODE REVIEW

**Comment by Gustl22** [comment]
Hmm I'm not sure about the update interval...we may should introduce a beta channel, so we don't have to bump the main version so frequently.
Besides I'm a big fan of regular updates, but keeping the breaking ones collected.

Also some Changelogs are duplicated, maybe melos messed sth up and compared to the wrong version(?)

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,79 @@
...
+ - **BREAKING** **REFACTOR**: rename logger_platform_interface.dart to global_platform_interface.dart ([#1385](https://github.com/bluefireteam/audioplayers/issues/1385)). ([6e837c1c](https://github.com/bluefireteam/audioplayers/commit/6e837c1ccd93b95d10843a403674128cf303c0ab))
+ - **BREAKING** **FIX**: remove unused `defaultToSpeaker` in `AudioContextIOS` and replace with `AVAudioSessionOptions.defaultToSpeaker` ([#1374](https://github.com/bluefireteam/audioplayers/issues/1374)). ([d844ef9d](https://github.com/bluefireteam/audioplayers/commit/d844ef9def06fd5047076d9f4c371ad3be4c8dd5))
+ - **BREAKING** **FIX**: Cache should take key to be properly cleared ([#1347](https://github.com/bluefireteam/audioplayers/issues/1347)). ([1a410bba](https://github.com/bluefireteam/audioplayers/commit/1a410bba578af506637b026bb2c4ace03a161a69))
+ - **BREAKING** **FEAT**: configurable SoundPool and `AudioManager.mode` ([#1388](https://github.com/bluefireteam/audioplayers/issues/1388)). ([5697f187](https://github.com/bluefireteam/audioplayers/commit/5697f187bcca64de2e519f8f49aaf4817fcf6398))
+
```
Comment (Gustl22) [inline CHANGELOG.md:42]: As far as I can tell, some of these changelogs were already in 2.0.0 or even 1.2.0. These duplicated changelogs are throughout all packages. We should remove them and only leave the ones which have actually changed from the previous release.

**Review by Gustl22 — CHANGES_REQUESTED** [review]
We should remove the duplicated changelogs before merging.

File: CHANGELOG.md (hunk 1 above)
Comment (spydon) [reply CHANGELOG.md:42]: Wtf, must have been a bug in melos, trying to fix it

**Comment by Gustl22** [comment]
> The problem is the maintenance cost that comes with that, since we'd have to have two branches that we cherry-pick back and forth from. But I do agree that we should try to collect breaking changes better than we have. Yesterday I just bumped it so that it would work with Flutter 3.7.0. 

I more thought of leaving the `main` branch as it is, but if there's need for a fast release, we first do a pre-release, that users can use which are dependent on it in that moment, and if it's more settled, we can release it as stable. So we save a few major releases in the stable release then. The users then cannot expect a reliable upgrade in the pre-release without breaking changes (as excpected), but get it earlier.

> There was probably something that accidentally changed within the same PR within that package then.

That was no accident. The flutter action upgraded the Flutter version at around the same time we made the release to be "compatible" with `3.7.0` except that it wasn't tested from our side, and so the lint is breaking.

**Comment by spydon** [comment]
> I more thought of leaving the `main` branch as it is, but if there's need for a fast release, we first do a pre-release, that users can use which are dependent on it in that moment, and if it's more settled, we can release it as stable. So we save a few major releases in the stable release then. The users then cannot expect a reliable upgrade in the pre-release without breaking changes (as excpected), but get it earlier.

That sounds good!

> That was no accident. The flutter action upgraded the Flutter version at around the same time we made the release to be "compatible" with `3.7.0` except that it wasn't tested from our side, and so the lint is breaking.

I mean that the versioning which is handled by melos picked up that there had been commits with `!` on those packages and therefore marked them as needing a major bump, but I think that was connected to the changelog bug that we just saw.

**Comment by Gustl22** [comment]
> I mean that the versioning which is handled by melos picked up that there had been commits with `!` on those packages and therefore marked them as needing a major bump, but I think that was connected to the changelog bug that we just saw.

Ah ok. Yes, I think so, too^^

**Comment by spydon** [comment]
Can you have a look at the changelogs again @Gustl22? :)
I hope I managed to get rid of the duplicates, but you know better what has been going in since it is almost all your PRs :sweat_smile:

**Review by Gustl22 — APPROVED** [review]

**Comment by Gustl22** [comment]
Perfect now! ;D Thank you!

**Comment by spydon** [comment]
> Perfect now! ;D Thank you!
> 
> Edit: except that mac doesn't like the new flutter version :/ maybe the example app has to be run once on a mac to update the mac specific config files.

Fixed it! It was a discontinued package (path_provider_macos) that was moved (to path_provider_foundation) but we had both in the pubspec.

**Comment by spydon** [comment]
Baah still something failing, any clue what it could be @Gustl22?

**Comment by Gustl22** [comment]
I try to run it on VM later, let's see 😅

**Comment by spydon** [comment]
> I try to run it on VM later, let's see 😅

Ran the example on my work mac and it seems to work fine, just a few of these in the logs when swapping to the stream tab:
```
2023-01-28 11:47:59.789 audioplayers_example[43120:3881330] Error: Message responses can be sent only once. Ignoring duplicate response on channel 'xyz.luan/audioplayers'.
```

Running the integration tests now, I've never done that locally before, it's quite a sound experience. 😂

**Comment by spydon** [comment]
Locally the MacOS integration tests passed on Ventura 13.1, which should be the same as the pipeline is running...

**Comment by spydon** [comment]
Apparently from Flutter 3.7.0 the macos build is by default render with metal, and GitHub CI doesn't support that. :cry:

**Comment by Gustl22** [comment]
> Apparently from Flutter 3.7.0 the macos build is by default render with metal, and GitHub CI doesn't support that. :cry: 

Is there already an issue open, and  maybe can we link that inside the code?

**Comment by spydon** [comment]
> > Apparently from Flutter 3.7.0 the macos build is by default render with metal, and GitHub CI doesn't support that. cry
> 
> Is there already an issue open, and maybe can we link that inside the code / Todo, so we don't loose track?

Added an issue and referenced it in the TODO.

**Comment by Gustl22** [comment]
Perfect! Finally getting there ;D

**Comment by spydon** [comment]
Phew!
I'll do a patch release of MacOS since there was something broken there. :)

**Pull request merged**